plugins/outputs/loki/stream.go

```diff
@@ -0,0 +1,66 @@
+package loki
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+)
+
+type (
+	Log []string
+
+	Streams struct {
+		Streams []Stream `json:"streams"`
+	}
+
+	Stream struct {
+		key string
+
+		Labels map[string]string `json:"stream"`
+		Logs   []Log             `json:"values"`
+	}
+)
+
+func (s *Streams) insertLog(ts []*telegraf.Tag, l Log) {
+	var (
+		key   = uniqKeyFromTagList(ts)
+		index int
+		found bool
+	)
+
+	for i, s := range s.Streams {
+		if s.key == key {
+			index, found = i, true
+			break
+		}
+	}
+
+	if !found {
+		s.Streams = append(s.Streams, newStream(key, ts))
+		index = len(s.Streams) - 1
+	}
+
+	s.Streams[index].Logs = append(s.Streams[index].Logs, l)
+}
+
+func uniqKeyFromTagList(ts []*telegraf.Tag) (k string) {
+	for _, t := range ts {
+		k += fmt.Sprintf("%s%s", t.Key, t.Value)
+	}
+
+	return
+}
```

Comment:
Are you really sure this is unique and correct? 

1. This will fail if tags are reordered but are otherwise the same. In this case you will generate a new "key" even though the tags are the same.
2. This will probably fail in case of overlapping key/value pairs in the tags. Assume a first metric that has `target=local, host=host, service=dns` and a second metric with `target=localhost, hostservice=dns`. With your "key-construction" you will get `targetlocalhosthostservicedns` in both cases and the metrics will be merged...

Is there a reason you don't use the metric ID that already exists (`metric.HashID()`)?

Reply:
`metric.HashID` seems to return an unique identifier for each metric, regarding is the same tags or not. I need the same identifier if the tags are the same

1. `[]*telegraf.Tag` are got from `TagList()` method, which mention tags are "by the tag key in lexical bytewise ascending order", seems to be good for me?
2. Indeed, I should improve this and the unit test

Reply:
point 2 should be good now